Onboarding Chunk 2

yougotwill opened the pull request:
~~Code review has been done on Will’s fork:~~
~~https://github.com/yougotwill/session-desktop/pull/43~~

Edit: Chunk 1 has been merged so we can do the final review here

yougotwill committed:
refactor: leftpanemessagesection is now a functional component

yougotwill committed:
feat: removed overlayheader

moved some parts into the leftpanesectionheader

yougotwill committed:
feat: added monospace textarea support to sessioninput

this will replace the sessionideditable eventually still needs error handling etc

yougotwill committed:
fix: restore copyOpenGroupURL key

it is still used

yougotwill committed:
feat: added some localised strings the create convo screens

yougotwill committed:
feat: improved error messaging when starting a new message

yougotwill committed:
feat: added conversation id to debugmessageinfo

yougotwill committed:
feat: closed group pubkeys cannot be used to create new messages

yougotwill committed:
feat: new message description and added zendesk link

yougotwill committed:
fix: correct type for style prop on SessionInput

yougotwill committed:
feat: update continue button styles

only show button when users enters something

yougotwill committed:
feat: updated style for choose action overlay

created new ActionRow component

yougotwill committed:
feat: updated ContactsListWithBreaks to show text when empty

yougotwill committed:
feat: consolidate remaining sessioneditable instances

fix styles for other overlays

yougotwill committed:
fix: cleaned up unused localised strings

since we swapped some toast messages for input messages

yougotwill committed:
feat: replace react-qr-svg with react-qrcode-logo

updated error correction to 25%

yougotwill committed:
feat: new sessionqrcode component

updated account id and recovery password qr codes with logos

yougotwill committed:
fix: the modal header now always centers content correctly

yougotwill committed:
feat: compressed qr logo images

yougotwill committed:
fix: add background colors to logos so the background matches

yougotwill committed:
fix: saved qr codes will be black and white by default

qr code logos can now be svgs for better rendering performance

yougotwill committed:
feat: invite a friend screen is mostly done

yougotwill committed:
feat: session input styles are now separated into per styling

monospaced, biggerText, isTextArea

yougotwill committed:
feat: finished invite a friend overlay

yougotwill committed:
Merge branch 'unstable' into feat/ses-825/onboarding2

yougotwill committed:
feat: log dumps for debugging restoring accounts

some config messages are not handled correctly and we need to know why

yougotwill committed:
fix: overlayInvite errors

yougotwill committed:
fix: was logging the push but not the dump

yougotwill committed:
fix: revert always on dump logging

yougotwill committed:
feat: show description text when making a message request

disabled conversation header functionality

yougotwill committed:
feat: disable composition for outgoing message requests

moved composition textarea to functional component

yougotwill committed:
feat: hide message preview in convo list item for outgoing requests

yougotwill committed:
fix: dont disable text area for outgoing message requests

we only want to disable some of the buttons

yougotwill committed:
feat: added updated group icon

add the ability to set custom icon sizes for some buttons

yougotwill committed:
feat: extracted password to new modal

added shiny hook to password protect anywhere

yougotwill committed:
fix: remove extra window

yougotwill committed:
fix: usemount imports have to be specific

otherwise we can get start errors due to an invalid import by react-use

yougotwill committed:
fix: make sure qr code fits the canvas contents

yougotwill committed:
feat: swapped out seed modal for settings category page

still work on category component but password protection works nicely

yougotwill committed:
fix: updated messageRequestPendingDescription copy with full stop

yougotwill committed:
feat: created new recovery phrase ui and qr code with toggling

also allow setting description to render html so we can have line breaks

yougotwill committed:
fix: make qr code update correctly if theme changes

even if visible, there is some flickering with the logo that I have been unable to fix.

yougotwill committed:
fix: improve qr code icon

fixed styling of qr icon on profile modal

yougotwill committed:
fix: recovery password text now uses primary text color for light modes

yougotwill committed:
feat: added icons to all settings categories

yougotwill committed:
feat: added hide password permenantly ui

yougotwill committed:
fix: brand logo in onboard header should not block buttons

yougotwill committed:
feat: added new hide recovery password setting

yougotwill committed:
feat: added new hide recovery password modal

when we turn on the setting we prevent any recovery phrase ui from loading and return null

yougotwill committed:
fix: removed SeedModal code

yougotwill committed:
fix: open recovery password settings page from left banner

yougotwill committed:
Merge branch 'unstable' into feat/ses-825/onboarding2

yougotwill committed:
fix: use localised strings for qr password toggle button

yougotwill committed:
feat: restrict importing useMount from react-use directly

it causes app start crashes

yougotwill committed:
feat: added mocharc.json

yougotwill committed:
feat: updated styled components and typings

cleaned up some extra SessionTheme declarations

yougotwill committed:
fix: added missing & selector to all pseudo events

yougotwill committed:
fix: repair global styling after styled upgrade improved match system setting

set each value on root instead of declaring one global css string. Use setThemeValues in all cases

yougotwill committed:
feat: improved zoom slider behaviour and styling

updated rc-slider, fixes issue where zoom happens before you release the mouse

yougotwill committed:
feat: added react testing library and chai dom

setup inital work for unit testing avatar placeholder bg color generation

yougotwill committed:
Merge branch 'unstable' into feat/ses-968/standardise_profile_picture

yougotwill committed:
fix: sessioninput color should be danger when there is an error

yougotwill committed:
fix: use recoveryPasswordOutline for light themes

updated to use building block svgs

yougotwill committed:
fix: replace setAccountPasswordTitle with lockApp string

yougotwill committed:
fix: use correct hero image

yougotwill committed:
fix: welcome message should use text-primary-color for light themes

yougotwill committed:
fix: repair placeholder selectors

required since upgrading styled-components

yougotwill committed:
fix: resolve cyclic imports with LokiProfile and MessageResultProps

yougotwill committed:
fix: resolve cyclic import with outgoing message type

avatar placeholder unit test now runs

yougotwill committed:
feat: added custom render function for react testing library

so we can load our own providers

yougotwill committed:
feat: addec typings for jsdom

can be useful is we want to parse html during unit testing

yougotwill committed:
fix: resolve cyclic imports in avatarplaceholder

bt moving types from data module identitykey, guardnode, snode, swarmnode to a separate file

yougotwill committed:
feat: added datatestids to avatarplaceholder and member placeholder

yougotwill committed:
feat: added print functions for react render results and html elements

used for unit testing

yougotwill committed:
feat: add component name support to printing functions

use prettyDom function for formatting instead of prettier directly

yougotwill committed:
feat: made good progress on avatar place holder unit test

feat added stubCrypto function

yougotwill committed:
feat: test for correct background color order is done

use libsodium for hash so we can unit test, remove crypto stub

yougotwill committed:
fix: there is already a unit test for initials

yougotwill committed:
feat: updated profile modal with style changes

cleaned up old sass and used style components instead

yougotwill committed:
feat: added flex gap support to Flex component

yougotwill committed:
fix: make sure larger modals fit the minimum window size

yougotwill committed:
fix: pass through events to onClose for modals

we can then use this to programmatically disable the onclose when necessary

yougotwill committed:
feat: created copy to clipboard button

yougotwill committed:
feat: added --font-size-xl

yougotwill committed:
feat: updated SessionInput to support multiple font sizes

this replaces the bigger text option, new props required, tabIndex, inputRef

yougotwill committed:
feat: spinner can now set width and height surrounding the svg

this is useful when we need it to take up less space

yougotwill committed:
fix: rename fontSize and disableOnBlur on Sessioninput

yougotwill committed:
feat: added SessionInput to Profile Modal

layout and style improvements, disable ui during name change, improved keyboard navigation

yougotwill committed:
feat: only center textarea placeholder text and not value in sessioninput

yougotwill committed:
feat: added filters to convo and contact query

search screen only shows contacts and not convos

yougotwill committed:
fix: contact list should not be case sensitive

yougotwill committed:
feat: added subtitle header to search results

yougotwill committed:
feat: search results are now sorted alphabetically without case sensitivity

also includes unknown contacts

yougotwill committed:
fix: clear search when change left pane modes

yougotwill committed:
fix: note to self always appears when there is a match regardless of case sensitivity

start search on first character since we limit the results

yougotwill committed:
fix: show no results text when search finds nothing

yougotwill committed:
fix: hide message preview for contact during search

yougotwill committed:
fix: message search results will use shortened pubkeys for authors

and will now resolve your blinded ids

yougotwill committed:
fix: borders on member list items and session radio

yougotwill committed:
feat: improved search for create group

yougotwill committed:
feat: improved about screen info

more info and copy to clipboard

yougotwill committed:
fix: reset right panel when we swap conversations

yougotwill committed:
feat: fixed update group name error and session recording light animations

yougotwill committed:
fix: typing animation alignment inside of the messages list

yougotwill committed:
feat: updated empty message image, typing animation and right panel slide durations

yougotwill committed:
feat: updated onion status and spinner aimations

yougotwill committed:
feat: updated message highlight durations

yougotwill committed:
fix: prevent the buttons from flashing when the app starts

yougotwill committed:
feat: updated onboarding container and progress bar

yougotwill committed:
feat: added copy to clipboard icon button

yougotwill committed:
fix: resolve outstanding react-use imports

yougotwill committed:
fix: quote icons were not appearing if media failed to create a preview

yougotwill committed:
feat: update invite a friend iniital string

yougotwill committed:
feat: sorted messages.json keys

yougotwill committed:
feat: added aria label support to buttons

yougotwill committed:
feat: added aria labels ot tos buttons in modal

yougotwill committed:
feat: account creation flow done

updated some testids and aria labels, SessionInput now has aria label support

yougotwill committed:
feat: account restoration flow done

updated some testids and aria labels, consolidated continue buttons into one component

yougotwill committed:
feat: recovery password settings flow done

added aria label and datatestid support to SessionQRCode

yougotwill committed:
feat: invite a friend flow done

consolidate help desk buttons into one component

yougotwill committed:
feat: profile modal flow done

refactored edit profile dialog and extracted out components and keyboard shortcuts

yougotwill committed:
fix: revert naming convo render functions as components

it broke the search input for some reason

yougotwill committed:
feat: added aria label to edit profile qr code

yougotwill committed:
feat: session input show hide button improvments

better aria labeling and dataTestIds

yougotwill committed:
fix: resolve typos

yougotwill committed:
Merge branch 'unstable' into feat/ses-825/onboarding2

yougotwill committed:
fix: remove duplicate debug key

yougotwill committed:
fix: isEmpty is not needed for searchTerm

because an empty string is falsy

yougotwill committed:
fix: properly type button and input ref

yougotwill committed:
fix: members list in the group overlay doesnt need a set bg color

removed !important from the transparent color assignment in the session radio

yougotwill committed:
fix: remove extra className prop

it is already included from the button types which we pass through the spread props

yougotwill committed:
fix: properly type send message input reference

this means we are properly typed for the composition mentions input and the paste handler

yougotwill committed:
fix: use input ref for password value instead of document lookup on pasword modal

yougotwill committed:
fix: replace keyup event listener with usekey hook

focus password input on mount for convenience

yougotwill committed:
fix: specify seconds in onion status glow duration comment

yougotwill committed:
fix: separate p tag styles into a component from styledprofilename

yougotwill committed:
fix: ses-2087 recovery password banner heading is too large

backported from chunk 1

yougotwill committed:
fix: ses-2089 update tos datatestid to open-url to match other platforms

backported from chunk 1 qa feedback

yougotwill committed:
fix: prevent font swap on first load

removed some unused sass classes, replace last sass font size variables

yougotwill committed:
fix: updated deblug log to use new font variable

yougotwill committed:
feat: added heading component

improved join community suggested rooms stylings

yougotwill committed:
fix: replace convo list render functions with components

yougotwill committed:
feat: updated hero image

backported from chunk 1 qa feedback

yougotwill committed:
fix: improved global window types for persistStore and inboxStore

we now await the persist store purge in deleteDbLocally

yougotwill committed:
feat: refactor theme state

added hooks for is light or dark theme and cleaned up any references

yougotwill committed:
fix: improvements to closedGroupOverlay including virtualize member list

createClosedGroupWithErrorHandling errorHandler changed to onError, simplify search checks

yougotwill committed:
fix: use i18n for continue button aria label

yougotwill committed:
fix: remove commented css on spinner

yougotwill committed:
fix: use i18n for aria labels on the enter display name button

yougotwill committed:
fix: sanitizeDisplayNameOrToast setDisplayNameError to onDisplayNameError

since it can be useState or redux

yougotwill committed:
fix: isSubtitle eval in search section header moved to parent

yougotwill committed:
fix: no need for reactnode or string type

reactnode supports strings

yougotwill committed:
fix: pick keys for icon prop in setting list item title since we dont use all of them

yougotwill committed:
fix: since the pass hash is in the items table we can call it from storage

added storage function getPasswordHash and use everywhere, warn user to restart the app

yougotwill committed:
fix: about window is now resizable

yougotwill committed:
feat: upgraded react-qrcode-logo to v3

yougotwill committed:
feat: updated shield icon and default colors

yougotwill committed:
feat: added motion props to AnimatedFlex

yougotwill committed:
fix: generate colors for the QR code in a hook that also converts svgs to data urls

we no longer need to update the svgs manually

yougotwill committed:
fix: resolve colors in convertIconToImageURL for qr code

yougotwill committed:
fix: first load flicker on qr code

yougotwill committed:
fix: pass resoled colors to session icon on first render

yougotwill committed:
feat: option to save qr codes with no colors by default

yougotwill committed:
fix: added skinny brand icon instead of messing with ratios

yougotwill committed:
fix: clean wip logs

yougotwill committed:
fix: move shortcuts back into edit profile

use region comments

yougotwill committed:
fix: move left pane header label into a  function

yougotwill committed:
feat: added new hot key hook since useKey doesnt ignore special keys

replaced all alpha numeric shortcuts

yougotwill committed:
fix: name showHide button props more explcitly

yougotwill committed:
fix: cleanup

yougotwill committed:
chore: left comment to use GenericWrapperActions.makeDump once it has been merged

yougotwill committed:
fix: remove unecessary length check on idsWithDisplayNames

yougotwill committed:
fix: dont export react testing library from renderComponent

prefer to import it explcitly when needed

yougotwill committed:
fix: suggest window log stubbing if it's not working when printing render or html elements

yougotwill committed:
fix: single line linux target in package json

otherwise appimages fail to build

yougotwill committed:
fix: search now works correctly and we include groups again

heading for unamed contacts is #, groups go above note to self, trust in the sql query

yougotwill committed:
fix: more review feedback

yougotwill committed:
fix: more review feedback part 2

yougotwill committed:
Merge branch 'unstable' into feat/ses-825/onboarding2

yougotwill committed:
Merge branch 'unstable' into feat/ses-825/onboarding2

bumped react and redux to the latest minior versions

yougotwill committed:
fix: correct admin crown color

yougotwill committed:
fix: play pause button in audio attachment

yougotwill committed:
feat: migrated lightbox to modal state

yougotwill committed:
feat: click on qr codes now opens the lightbox

lightbox now supports saving data blobs

yougotwill committed:
feat: move db deletion methods to accountManager

yougotwill committed:
feat: remove modal confirm button container

use no padding class on modal instead, less messing around with ghost buttons in certain cases

yougotwill committed:
feat: added quit modal

when going back during critical onboarding steps it's better to restart the app entirely

yougotwill committed:
feat: added updateOurProfileDisplayName

which uses libsession to check the display name length, not yet used in onboarding

yougotwill committed:
feat: start of using updateOurProfileDisplayName in onboarding

yougotwill committed:
fix: correctly trim usernames

yougotwill committed:
feat: use CONTACT_MAX_NAME_LENGTH

as max length for session inputs for user profile name

yougotwill committed:
fix: truncate submitted display names

we dont look for libsession throws when setting a display name.
libsession now truncates by default

yougotwill committed:
fix: formatting fixes

use profile manager name validation in CreateAccount component

yougotwill committed:
feat: commment on libsession constants

give context that CONSTANTS is not a object but an immediately invoked
function

yougotwill committed:
feat: added error boundary to component rendering in unit tests, resolve issues SVGElement in JSDOM

yougotwill committed:
fix: remove unneeded catch clause in profilemanager

yougotwill committed:
fix: replace chai with react-test-renderer for unit testing components

rewrite avatar placeholder test

yougotwill committed:
fix: remove unneeded catch

Bilb commented on ts/components/registration/components/BackButton.tsx:
```diff
@@ -48,6 +76,36 @@ export const BackButton = ({ callback }: { callback?: () => void }) => {
       iconRotation={90}
       padding={'0'}
       onClick={() => {
+        if (shouldQuit && quitMessage) {
```
maybe rename this props here and in other places to be shouldQuitOnClick

Bilb commented on ts/components/registration/stages/CreateAccount.tsx:
```diff
@@ -77,6 +82,7 @@ export const CreateAccount = () => {
       const newHexPubKey = StringUtils.decode(keyPair.pubKey, 'hex');
 
       dispatch(setRecoveryPassword(mnemonic));
+      dispatch(setPrivateKeyBytes(keyPair.ed25519KeyPair.privateKey));
```
idk if that's uint8array or not, but redux doesn't like it if yes. Can you store it as hex instead (sodium exposes from_hex/to_hex)

Bilb commented on ts/components/dialog/edit-profile/EditProfileDialog.tsx:
```diff
@@ -226,11 +212,17 @@ export const EditProfileDialog = () => {
-    setLoading(false);
+    try {
+      setLoading(true);
+      const validName = await ProfileManager.updateOurProfileDisplayName(profileName);
+      setUpdateProfileName(validName);
+      setMode('default');
+    } catch (err) {
+      window.log.error('Profile update error', err);
+      setProfileNameError(window.i18n('unknownError'));
+    } finally {
+      setLoading(false);
+    }
```
is the error found by gh action here a false positive?

![image](https://github.com/user-attachments/assets/bfb6ae7c-37d5-4f31-86ca-7e98d07f25b7)

Bilb replied:
don't see an issue so I guess we are fine

Bilb commented on ts/components/lightbox/LightboxGallery.tsx:
```diff
@@ -23,23 +24,27 @@ export interface MediaItemType {
 export const LightboxGallery = (props: Props) => {
-  const { media } = props;
+  const { media, selectedIndex = -1, onClose } = props;
   const [currentIndex, setCurrentIndex] = useState(-1);
   const selectedConversation = useSelectedConversationKey();
 
   const dispatch = useDispatch();
 
   // just run once, when the component is mounted. It's to show the lightbox on the specified index at start.
   useEffect(() => {
-    setCurrentIndex(props.selectedIndex);
+    setCurrentIndex(selectedIndex);
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
```
could this not me `const [currentIndex, setCurrentIndex] = useState(props.selectedIndex)` ?

Bilb commented on ts/util/accountManager.ts:
```diff
@@ -236,3 +244,153 @@ export async function registrationDone(ourPubkey: string, displayName: string) {
+      await deleteDbLocally();
+    } catch (e) {
+      window?.log?.error(e);
+    } finally {
+      window.restart();
+    }
```
this could actually be made a finally on the outer try/catch, but it's probably easier to read like this

Bilb commented on ts/util/accountManager.ts:
```diff
@@ -236,3 +244,153 @@ export async function registrationDone(ourPubkey: string, displayName: string) {
+    if (allRoomInfos && allRoomInfos.size > 0) {
+      // clear each inbox per sogs
+      // eslint-disable-next-line no-restricted-syntax
+      for (const roomInfo of allRoomInfos.values()) {
+        try {
+          // eslint-disable-next-line no-await-in-loop
+          const success = await clearInbox(roomInfo);
+          if (!success) {
+            throw Error(`Failed to clear inbox for ${roomInfo.conversationId}`);
+          }
+        } catch (error) {
+          window.log.info('DeleteAccount =>', error);
+          continue;
+        }
```
we should maybe Promise.allSettled those calls to make it faster

Bilb commented on ts/session/constants.ts:
```diff
@@ -72,15 +72,13 @@ export const MAX_ATTACHMENT_FILESIZE_BYTES = 10 * 1000 * 1000;
 
 export const VALIDATION = {
   MAX_GROUP_NAME_LENGTH: 30,
+  MAX_COMMUNITY_NAME_LENGTH: 300,
   CLOSED_GROUP_SIZE_LIMIT: 100,
 };
```
or get rid of those entirely?

yougotwill committed:
fix: aded simple test for sessioninput

made note to migratie to storybook in the future

yougotwill committed:
fix: update translation link

show external link on hover

yougotwill committed:
Merge branch 'feat/no-ref/unit_test_sessioninput' into feat/ses-825/onboarding2

yougotwill committed:
feat: updated turkish translations

See https://github.com/oxen-io/session-desktop/pull/3122

yougotwill committed:
fix: review feedback - lightbox defaults

yougotwill committed:
feat: convert privKeyBytes to hex

redux doesnt like uint8Array, rename shouldQuit to shouldQuitOnClick

yougotwill committed:
fix: review feedback - unmount qr generated with finally on save

yougotwill committed:
fix: move finally to outer try catch for delete method

yougotwill committed:
chore: update macOS runner to 12

11 has been deprecated

yougotwill committed:
fix: replace MAX_COMMUNITY_NAME_LENGTH

using libsession COMMUNITY_FULL_URL_MAX_LENGTH

yougotwill committed:
fix: replace MAX_GROUP_NAME_LENGTH

using libsession BASE_GROUP_MAX_NAME_LENGTH

yougotwill committed:
fix: error test for input should be selectable

makes error reporting easier

yougotwill committed:
fix: disable join button when already joining a community

yougotwill committed:
feat: use promise all settled for clear sogs inboxes

yougotwill committed:
feat: use testkey for user profile update in onboarding for testing a display name

remove privateKeyBytesHex since we don't use it

yougotwill reviewed (commented):
Responded

yougotwill committed:
chore: bump libsession_util_nodejs to v0.3.20

yougotwill committed:
Merge branch 'unstable' into feat/ses-825/onboarding2

yougotwill committed:
fix: a whole bunch of circular depdencies

for various convo duck, convo interactions, convo model and opengroup types

yougotwill committed:
fix: ran formatter

yougotwill committed:
fix: guard node test import order

yougotwill committed:
fix: fix updateOurProfileDisplayName breaking tests

don't use testing functions to generate a temp secret key rather do it manually

yougotwill committed:
fix: overflow in session modal title

user details dialog is now a fixed width

yougotwill committed:
fix: remove uneeded cleanup() calls in component unit tests

Bilb committed:
fix: failing tests making requests to real servers :bad:

yougotwill committed:
fix: remove stubstorage

yougotwill committed:
fix: no overflow in group members list in overlay

Bilb commented on ts/components/registration/components/ContinueButton.tsx:
```diff
@@ -0,0 +1,21 @@
+
+  return (
+    <SessionButton
+      ariaLabel={window.i18n('continue')}
```
I meant the arialabels. Not all of them are localized

yougotwill replied:
I don't think we are localising all aria labels yet? Just coincidence that I had the right word to use. I believe that is a separate future project. Assuming you are referring to https://github.com/yougotwill/session-desktop/blob/5a68985f4a08bd85d9faa392d0cfe669daec5e23/ts/components/leftpane/LeftPaneSectionHeader.tsx#L162 which is the only one I could find

Bilb replied:
ok all good

Bilb commented on ts/components/lightbox/LightboxGallery.tsx:
```diff
@@ -30,16 +31,16 @@ type Props = {
     setCurrentIndex(selectedIndex);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  });
```
is this needed at all?

yougotwill replied:
Good catch!

Bilb commented on ts/components/registration/stages/CreateAccount.tsx:
```diff
@@ -97,10 +93,6 @@ export const CreateAccount = () => {
     try {
-      if (!privateKeyBytes) {
-        throw new Error('Private key not found');
-      }
```
you've changed this so all good, but just so you know an empty array is truthy
![image](https://github.com/user-attachments/assets/3f972958-8d4c-4152-87f3-dfefcebcbc16)

yougotwill replied:
Yup I know I probably messed up here previously

Bilb commented on ts/data/opengroups.ts:
```diff
@@ -14,35 +14,35 @@ export type OpenGroupV2RoomWithImageID = {
+// export type OpenGroupV2Room = {
+//   serverUrl: string;
 
-  /** this is actually shared for all this server's room */
-  serverPublicKey: string;
-  roomId: string;
+//   /** this is actually shared for all this server's room */
+//   serverPublicKey: string;
+//   roomId: string;
```
can be removed altogether? You sure we don't need this anywhere?

Bilb replied:
all good, you moved it somewhere else, so you can remove it from here

Bilb commented on ts/test/components/AvatarPlaceHolder_test.tsx:
```diff
@@ -56,71 +46,33 @@ describe('AvatarPlaceHolder', () => {
+    const result = renderComponent(
+      // NOTE we need to test the pubkey to color generation and ordering with appium. Since we can't access the value of a css variable in with the current unit test setup
+      <AvatarPlaceHolder
+        diameter={AvatarSize.XL}
```
can we not at least verify that the string var(--color-xxx) is the right one for each pubkeys?

yougotwill replied:
Unfortunately not because it possible from the component because the color is just the primary color variable. I wanted avoid abstrating the color generation into a function. Which is what strated unit testing components in the first place. I can do that and then we can test the individual color values.

i.e. The color prop returned is `var(--primar-color)` it does not resolve to a hex code

Bilb commented on ts/util/accountManager.ts:
```diff
@@ -302,21 +301,31 @@ export async function deleteEverythingAndNetworkData() {
+          window.log.info('Inbox cleared for room', allRoomInfosArray[index]);
         }
-      }
+      });
```
you shouldn't have the return {status: fullfilled/rejected by hand). You should just have to return something/throw in the promise, and Promise.allSettled will report the ones which failed as status === "rejected" and others as fullfilled.

yougotwill replied:
Fixed!

yougotwill committed:
fix: revert virtualized group members list

since it caused overflow, disabled buttons when making a group

yougotwill committed:
feat: fixed member layout in invite contancts and remove mods modals

yougotwill committed:
feat: fixed member layout in update group modals

yougotwill committed:
fix: user mentions overflow

yougotwill committed:
fix: blocked list layout and list rendering overflow

yougotwill committed:
fix: review feedback

unneeded promise object returned

yougotwill committed:
fix: styled name prop should just be max width

Bilb commented on ts/components/DebugLogView.tsx:
```diff
@@ -42,8 +42,9 @@ const StyledContent = styled.div`
     min-height: 100px;
 
-    font-family: Monaco, Consolas, 'Courier New', Courier, monospace;
+    font-family: var(--font-debug);
```
fine for  me

Bilb commented on ts/state/ducks/conversations.ts:
```diff
@@ -1184,6 +1174,7 @@ export async function openConversationWithMessages(args: {
       initialQuotes,
     })
   );
+  window.inboxStore?.dispatch(resetRightOverlayMode());
```
it used to. Is this something you've changed? 
![image](https://github.com/user-attachments/assets/3b3dbd1b-b6f6-4534-9362-497975921ef0)

Bilb replied:
oh yeah, my bad I got it
